tests/test_schema.py

```diff
@@ -47,3 +48,47 @@ def test_schema_size(self):
 
         _tuple = {"type": "array", "items": [{"type": "string"}, {"type": "string"}]}
         assert Schema._get_schema_size(_tuple) == 2
+
+    def test_set_elem_propogate(self):
+        schema = Schema({
+            "properties": {
+                "a": {
+                    "type": "int"
+                }
+            },
+            "type": "object"
+        })
+
+        res_schema = schema.clone()
+
+        # Shouldn't add it if not propogating
+        res_schema.set_schema_elem(
+                ("properties", "b", "properties", "hello"),
+                {"type": "string"},
+                propogate=False)
+
+        print_and_test(schema.schema, res_schema.schema)
+
+        # should add it if propogating
+        expected = {
+            "properties": {
+                "a": {
+                    "type": "int"
+                },
+                "b": {
+                    "properties": {
+                        "hello": {
+                            "type": "string"
+                        },
+                    },
+                    "type": "object"
+                }
+            },
+            "type": "object"
+        }
+
+        res_schema.set_schema_elem(
+                ("properties", "b", "properties", "hello"),
+                {"type": "string"},
+                propogate=True)
+        print_and_test(expected, res_schema.schema)
```

Comment:
When `properties.b.properties.hello` is removed, does this turn back into `schema.schema`?

Reply:
Good question, it should if `propogate=True` for the deletion. I'll add this.